hasher-matcher-actioner/terraform/authentication-shared/main.tf

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+resource "aws_cognito_user_pool" "webapp_and_api_shared_user_pool" {
```

Comment:
do you know a way we can reduce the copy paste from not shared?

Reply:
No, sorry. I figured this comment would come up - and it's a good one. I tried to think of something that would help with that. The issue is that I wanted to keep the names of the user pool resources different so terraform state wouldn't get messed up in some way. I searched terraform docs for the concept of re-use at the resource level and couldn't find anything (like using a var in the resource name label).

Reply:
can we have name labels uniqueness from the module level? e.g. we could have a wrapper module here that way it would be `shared_pool.webapp_and_api_shared_user_pool` vs `authentication.webapp_and_api_shared_user_pool`

Reply:
Done